sycl/test/usm/allocator_vector_fail.cpp

```diff
@@ -0,0 +1,55 @@
+// RUN: %clang -std=c++11 -fsycl %s -o %t1.out -lstdc++ -lOpenCL -lsycl
+// RUN: %CPU_RUN_PLACEHOLDER %t1.out
+// XFAIL: *
```

Comment:
I'll add the comment.  The test is supposed to fail b/c device pointers cannot be dereferenced on the host.  The test is designed to make the allocator throw an exception because it's using device pointers.  Would you rather the test catch the exception and report passing if it does?

Reply:
I would rather catch the exception and check that it's exactly what we expect.
We should have 0 tests that expected to fail.